src/widgets/table/table.rs

```diff
@@ -1248,12 +1240,17 @@ mod tests {
             );
 
             // with selection, less than needed width
+            // <--------7px-------->
+            // ┌────────┐x┌────────┐
+            // │ (3, 3) │x│ (7, 0) │
+            // └────────┘x└────────┘
+            // column spacing (i.e. `x`) is always prioritized
             test(
                 &[Length(4), Length(4)],
                 SegmentSize::None,
                 7,
                 3,
-                &[(3, 4), (7, 0)],
+                &[(3, 3), (7, 0)],
             );
```

Comment:
This is an example where using DRY for tests makes the tests harder to understand, not easier.
```suggestion
            // with selection, less than needed width
            // <--------7px-------->
            // ┌────────┐x┌────────┐
            // │ (3, 3) │x│ (7, 0) │
            // └────────┘x└────────┘
            // column spacing (i.e. `x`) is always prioritized
            let table = Table::default()
                .widths([Length(4), Length(4)])
                .segment_size(SegmentSize::None);
            assert_eq!(table.get_columns_widths(7, 3), [(3, 3), (7, 0)]);
```

Reply:
fixed in 3475ea92409b03fb1a9ac2bc3f41f3889b4f402d

Reply:
and 161492f1996b5af1042e2cf1104343832d3c07c2